config/config.go

```diff
@@ -746,6 +748,32 @@ func (re *Regexp) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+// Matchers encapsulates labels.Matchers and makes it YAML marshalable.
+type Matchers labels.Matchers
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface for Matchers.
+func (m *Matchers) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	var lines []string
+	if err := unmarshal(&lines); err != nil {
+		return err
+	}
+	for _, line := range lines {
+		labelMatchers, err := labels.ParseMatchers(line)
```

Comment:
Not really happy with `labelMatchers`. How about `parsedMatchers`? Since the scope is so narrow here, you could even go for something like `mm` or `ms` or `pm`.